### PR TITLE
Polish the README and move data conversion table to the docs

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,16 +31,20 @@ term = Msgpax.unpack!(iodata)
 # => [300, "Spartans"]
 ```
 
-#### Stream-oriented deserialization
+#### Partial deserialization
```
Comment (lexmag) [inline README.md]: If we're going with "Partial" (seems better than old) we should do the same change in `unpack_slice` doc. :bowtie:

Comment (whatyouhide) [inline README.md]: Good point.

**Comment by whatyouhide** [comment]
@lexmag fixed the "stream-oriented" thing in the latest commit.

**Comment by lexmag** [comment]
:yellow_heart:

**Pull request merged**